### PR TITLE
Necessary changes to make it work

### PR DESCRIPTION
**Pull request by reakaleek**

### CODE REVIEW

**Review by v1v — APPROVED** [review]

File: lib/setup.bash [hunk 1]
```diff
@@ -26,15 +26,7 @@ function setup() {
 	local -r asset_id=$(get_asset_id "$version")
 	mkdir -p "${bin_dir}"
 	download_asset "$asset_id" "$bin_dir"
-
-	export PATH="${bin_dir}:${PATH}"
-	export GITHUB_TOKEN="${GH_TOKEN}"
-	export ELASTIC_APM_ENVIRONMENT="ci"
-
-	git config --global user.name "${GIT_USER:-"${username}"}"
```
Comment (v1v) [inline lib/setup.bash:34]: what's the reason for this change?

Comment (reakaleek) [reply lib/setup.bash:34]: Not sure. I am still trying to figure out what's best.

I got it working in https://github.com/elastic/apm-managed-service/pull/890

But I needed to run `gh auth setup-git`. And didn't want to add it here. Not sure.

Unless I just add `gh` as a requirement. 

But in the e2e pipeline of this repo `gh auth setup-git` is not needed.

Comment (reakaleek) [reply lib/setup.bash:34]: Also, we are not doing the setup-git step in bolt-actions aswell, there it's also a different action doing it before oblt-cli setup.

Comment (v1v) [reply lib/setup.bash:34]: Do you think we can rename `setup.bash` for `install.bash`?

File: plugin.yml [hunk 2]
```diff
@@ -7,7 +7,6 @@ requirements:
   - curl
   - bash
   - tar
```
Comment (reakaleek) [inline plugin.yml:9]:
```suggestion
  - tar
  - git
```

File: hooks/pre-command [hunk 3]
```diff
@@ -4,15 +4,29 @@ set -euo pipefail
...
+. "${DIR}/../lib/setup.bash"
+
+git config --global user.name "${GIT_USER:-"${OBLT_CLI_USERNAME}"}"
+git config --global user.email "${GIT_EMAIL:-"${OBLT_CLI_USERNAME}@users.noreply.github.com"}"
+
+git config --global credential.helper store
+echo "protocol=https
+host=github.com
+username=token
+password=$OBLT_CLI_GH_TOKEN" | git credential-store store
```
Comment (v1v) [inline hooks/pre-command:28]: Should a pre-exit hook be included too and delete this entry? Just wanna be sure static workers don't have secrets in plain text

Comment (reakaleek) [reply hooks/pre-command:28]: https://github.com/elastic/oblt-cli-buildkite-plugin/pull/7/commits/ab0d60ca88d616bcfb2025f25571b00294f92be5

Comment (v1v) [reply hooks/pre-command:28]: I don't know if you need this, I've been searching how the existing Buildkite/Vault integration works when provisioning BK runners:

- see [this](https://github.com/elastic/vault-buildkite-plugin/blob/d2f85f85efdf229b341c3653938a0c64d792a6a5/hooks/environment#L314) (Only accessible by  Elastic employees)

Comment (reakaleek) [reply hooks/pre-command:28]: You mean, we can revert the commit?

Comment (v1v) [reply hooks/pre-command:28]: I think so, the existing BK/Vault integration configures the username/password using the GitHub ephemeral token. And if we change it then it might cause some issues?

I wonder whether  the `OBLT_CLI_GH_TOKEN` might not be needed at all and and we will need to rely on the `VAULT_GITHUB_TOKEN` and the permissions-set configuration, what do you think?

Comment (reakaleek) [reply hooks/pre-command:28]: The limitation of the permission set worries me. Because of oblt-cli, we need to change the `contents: read` permission to `contents: write`.

But I agree that if we only rely on the `VAULT_GITHUB_TOKEN`, it will have the most native experience and no surprises because of the modified git credentials.

Comment (reakaleek) [reply hooks/pre-command:28]: Revert commit: https://github.com/elastic/oblt-cli-buildkite-plugin/pull/7/commits/79ec172945d4a7fb7999ce2098951f00b025612c

Comment (reakaleek) [reply hooks/pre-command]: Updated in https://github.com/elastic/oblt-cli-buildkite-plugin/pull/9

**Review by v1v — APPROVED** [review]
